src/webgl/shaders/line.frag

```diff
@@ -1,8 +1,11 @@
 precision mediump float;
 precision mediump int;
 
+varying vec4 vColor;
+
+uniform bool uUseLineColor;
 uniform vec4 uMaterialColor;
 
 void main() {
-  gl_FragColor = uMaterialColor;
-}
\ No newline at end of file
+  gl_FragColor = (uUseLineColor ? vColor : uMaterialColor);
```

Comment:
We can maybe avoid doing this in the fragment shader if, in the vertex shader, we set `vColor` as either `uUseLineColor ? aVertexColor : uMaterialColor`. This is a pretty insignificant optimization, but there are usually more fragments than vertices, so it's marginally faster to do if statements and such in the vertex shader and do less checks in the fragment shader (but again, it's just one if statement, so this is more for "best practices" than for a measurable gain)

Reply:
I see, the vertexShader receives uMaterialColor and writes the branch processing...certainly, considering the number of calls, that might be better...Okay, I'll give it a try!